server/db/driver/pg/orders.go

```diff
@@ -57,7 +58,13 @@ func (coins *dbCoins) Scan(src interface{}) error {
 		if len(b) < cLen+1 {
 			return fmt.Errorf("too many bytes indicated")
 		}
-		c = append(c, b[1:cLen+1])
+
+		// Deep copy the coin ID (a slice) since the backing buffer may be
+		// reused.
+		bc := make([]byte, cLen)
+		copy(bc, b[1:cLen+1])
+		c = append(c, bc)
+
```

Comment:
This was subtle and hard to catch, but the lib/pq driver did hit it and the backing buffers were reused.